### PR TITLE
Fixes #89 - Mongo test controller

### LINKED ISSUE #89: Create a Mongo pre-enabled AbstractCRUDTestController
This is from a conversation with @paterczm that happened over in: https://github.com/lightblue-platform/lightblue-test-utils/pull/6

Essentially the idea is that we want to make a nice fluid experience for mongo backend users as that is the common (only) use case and we want it to be as simple as possible.

### PR DESCRIPTION
**Pull request by dcrissman**
Adds an implementation of AbstractCRUDTestController that has Mongo support built in, as that is the only supported metadata backend at this time.

### CODE REVIEW

**Comment by dcrissman** [comment]
Fixes https://github.com/lightblue-platform/lightblue-mongo/pull/92

**Comment by dcrissman** [comment]
Fixes #89

**Comment by dcrissman** [comment]
This change depends on https://github.com/lightblue-platform/lightblue-core/pull/356

**Comment by dcrissman** [comment]
Ready for review

File: test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    public AbstractMongoCRUDTestController() throws Exception {
+        super(true);
+    }
+
+    @Override
+    protected JsonNode getDatasourcesJson() {
+        try {
+            if (getDatasourcesResourceName() == null) {
+                return json(loadResource("/mongo-datasources.json", true));
```
Comment (jewzaam) [inline test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java]: Should the resource be `./mongo-datasources.json` instead?

File: test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    @BeforeClass
+    public static void prepareMongoDatasources() {
+        if (System.getProperty("mongo.host") == null) {
+            System.setProperty("mongo.host", "localhost");
+        }
+        if (System.getProperty("mongo.port") == null) {
+            System.setProperty("mongo.port", String.valueOf(mongoServer.getPort()));
+        }
+        if (System.getProperty("mongo.database") == null) {
+            System.setProperty("mongo.database", "lightblue");
```
Comment (jewzaam) [inline test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java]: Why pick a default database name of "lightblue"?  I think we've used "mongodata" or "data" elsewhere, so curious.

Comment (dcrissman) [inline test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java]: It is the value I have used elsewhere, but really I just threw the value in. I am happy to use something else if you'd like.

File: test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java (hunk 1 above)
Comment (dcrissman) [inline test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java]: Honestly, I am not sure. Both ways seem to work. I followed @paterczm (name drop alert) example  over in https://github.com/lightblue-platform/lightblue-core/blob/master/test/src/main/java/com/redhat/lightblue/test/AbstractCRUDTestController.java#L108

I am happy to go either direction.

Comment (jewzaam) [inline test/src/main/java/com/redhat/lightblue/mongo/test/AbstractMongoCRUDTestController.java]: Guess it doesn't matter if it works.  I've always done the `./` prefix.

**Pull request merged**